sql/sql_optimizer.cc

```diff
@@ -10162,16 +10162,29 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
 
       if (new_item == NULL)
       {
+        bool replace_always_true = false;
         if (is_cond_or)
         {
-          Item* cond_true =
-            new (thd->mem_root) Item_int((longlong)1, 1); // Always true
-          (void)li.replace(cond_true);
-          should_fix_fields = true;
+          if (dynamic_cast<Item_int*>(item))
```

Comment:
if (item->type() == Item::INT_ITEM) {
   convert to Item_int; then do something else. 
}
else if (item->type() == ITEM::FUNC_ITEM  {
   convert to Item_bool_func2; then do something else. otherwise,  the conversion will failed or damage the memory. 
}

Reply:
that's good